src/BookmarkEditor/component.tsx

```diff
@@ -1,26 +1,37 @@
 import './component.css'
-import { removeBookmark, updateBookmark } from '../Bookmarks/repository'
-import { Bookmark } from '../Bookmarks/model'
+import { Bookmark, shortcutKeyOf } from '../Bookmarks/model'
+import { removeBookmark, useShortcutMap } from '../Bookmarks/repository'
+import { EditingBookmark } from './model'
 import { FC } from 'react'
 
 type BookmarkEditorComponentProps = {
-  bookmark?: Bookmark
+  editingBookmark?: EditingBookmark
   onChange: (newValue: Bookmark) => void
   onRequestClose: () => void
 }
 
-const BookmarkEditorComponent: FC<BookmarkEditorComponentProps> = ({ bookmark, onChange, onRequestClose }) => {
-  if (bookmark === undefined) {
+const BookmarkEditorComponent: FC<BookmarkEditorComponentProps> = ({ editingBookmark, onChange, onRequestClose }) => {
+  const [shortcutMap, setShortcutMap] = useShortcutMap()
+  if (editingBookmark === undefined) {
     return null
   }
   return (
     <div>
       <div className="BookmarkEditor__Modal">
         <FormComponent
-          bookmark={bookmark}
+          bookmark={editingBookmark}
           onChange={onChange}
-          onSubmit={() => void updateBookmark(bookmark).then(() => onRequestClose())}
-          onRemove={() => void removeBookmark(bookmark).then(() => onRequestClose())}
+          onSubmit={() => {
```

Comment:
This should call `updateBookmark`. Fixed in https://github.com/int128/bntp/pull/106.